### PR TITLE
Executor and Planner for IndexScan

### PR DESCRIPTION
**Pull request by bright-starry-sky**
From PR #161.

### CODE REVIEW

File: src/planner/Query.h [hunk 1]
```diff
@@ -439,10 +439,105 @@ class GetEdges final : public Explore {
...
+
+    static IndexScan* make(ExecutionPlan* plan, PlanNode* input, GraphSpaceID space) {
+        return new IndexScan(plan, input, space);
+    }
+
+    static IndexScan* make(ExecutionPlan* plan,
+                           PlanNode* input,
+                           GraphSpaceID space,
+                           IndexQueryCtx&& contexts,
+                           IndexReturnCols&& returnCols,
```
Comment (Shylock-Hg) [inline src/planner/Query.h:453]: Is this duplicate with the field colNames_?

Comment (bright-starry-sky) [reply src/planner/Query.h:453]: > Is this duplicate with the field colNames_?

Maybe yes. Let's leave it at that for now, whether change is needed depends on the validator.

File: src/executor/query/IndexScanExecutor.cpp [hunk 2]
```diff
@@ -7,15 +7,91 @@
...
+                                      lookup->isEdge(),
+                                      lookup->schemaId(),
+                                      *lookup->returnColumns())
+        .via(runner())
+        .then([this](StorageRpcResponse<LookupIndexResp> &&rpcResp) {
+            return handleResp(std::move(rpcResp));
+        });
+}
+
+template <typename Resp>
```
Comment (Shylock-Hg) [inline src/executor/query/IndexScanExecutor.cpp:37]: This template is useless.

**Comment by dutor** [comment]
Please resolve the conflicts.

**Review by dutor — APPROVED** [review]
LGTM.

**Review by dutor — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**